pysb/pattern.py

```diff
@@ -1,25 +1,27 @@
 import collections
 from .core import ComplexPattern, MonomerPattern, Monomer, \
-    ReactionPattern, ANY, as_complex_pattern
+    ReactionPattern, ANY, as_complex_pattern, DanglingBondError
 import networkx as nx
 from networkx.algorithms.isomorphism.vf2userfunc import GraphMatcher
 from networkx.algorithms.isomorphism import categorical_node_match
 import numpy as np
 
 
-def get_bonds_in_pattern(pat):
+def get_bonds_in_pattern(pat, as_set=True):
```

Comment:
This new interface seems unwieldy. A suggestion: Rename this function to `get_half_bonds_in_pattern(pat)` and have it always return the `list`. Then implement `get_bonds_in_pattern` as:
```python
def get_bonds_in_pattern(pat):
    return set(get_half_bonds_in_pattern(pat))
```
Then `check_dangling_bonds` would call `get_half_bonds_in_pattern`.